cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/AwsInstanceConnector.java

```diff
@@ -327,13 +327,15 @@ private List<CloudVmInstanceStatus> fillCloudVmInstanceStatuses(AuthenticatedCon
         return cloudVmInstanceStatuses;
     }
 
-    private Collection<String> instanceIdsWhichAreNotInCorrectState(List<CloudInstance> vms, AmazonEc2Client amazonEC2Client, String state) {
+    private Collection<String> instanceIdsWhichAreNotInCorrectState(List<CloudInstance> vms, AmazonEc2Client amazonEC2Client, String state,
+            Set<InstanceStatus> completedStatuses) {
         Set<String> instances = vms.stream().map(CloudInstance::getInstanceId).collect(Collectors.toCollection(HashSet::new));
         DescribeInstancesResponse describeInstances = amazonEC2Client.describeInstances(
                 DescribeInstancesRequest.builder().instanceIds(instances).build());
+        Set<String> status = completedStatuses.stream().map(i -> i.toString().toLowerCase()).collect(Collectors.toUnmodifiableSet());
         for (Reservation reservation : describeInstances.reservations()) {
             for (Instance instance : reservation.instances()) {
-                if (state.equalsIgnoreCase(instance.state().name().toString())) {
+                if (state.equalsIgnoreCase(instance.state().name().toString()) || status.contains(instance.state().name().toString())) {
```

Comment:
status always have strings in lower case but we are not converting instance.state().name().toString() to lower case so it does not seem like case insensitive match. Please check.

Reply:
The "Running" and "Stopped" are in this manner and all the other states are in lowercase by default from the aws end.
So, this should not be a problem

Reply:
the instance state coming from the cloud provider end which is aws is always in lower case so I have to transform the one from our end to lower case.
So, it is not a problem